src/logic/safe/store/actions/fetchSafe.ts

```diff
@@ -1,178 +1,187 @@
+import { BalanceEndpoint, fetchTokenCurrenciesBalances } from 'src/logic/safe/api/fetchTokenCurrenciesBalances'
+import { getLocalSafe } from 'src/logic/safe/utils'
 import { enabledFeatures, safeNeedsUpdate } from 'src/logic/safe/utils/safeVersion'
-import { sameAddress } from 'src/logic/wallets/ethAddresses'
-import { getBalanceInEtherOf } from 'src/logic/wallets/getWeb3'
-import { addSafeOwner } from 'src/logic/safe/store/actions/addSafeOwner'
-import { removeSafeOwner } from 'src/logic/safe/store/actions/removeSafeOwner'
+import { fromTokenUnit } from 'src/logic/tokens/utils/humanReadableValue'
+import { sameAddress, ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
 import updateSafe from 'src/logic/safe/store/actions/updateSafe'
 import { makeOwner } from 'src/logic/safe/store/models/owner'
 import { checksumAddress } from 'src/utils/checksumAddress'
 import { SafeOwner, SafeRecordProps } from 'src/logic/safe/store/models/safe'
-import { AppReduxState } from 'src/store'
-import { latestMasterContractVersionSelector, safeTotalFiatBalanceSelector } from 'src/logic/safe/store/selectors'
-import { getSafeInfo } from 'src/logic/safe/utils/safeInformation'
+import { getSafeInfo, SafeInfo } from 'src/logic/safe/utils/safeInformation'
 import { getModules } from 'src/logic/safe/utils/modules'
 import { getSpendingLimits } from 'src/logic/safe/utils/spendingLimits'
 import { LOADED_SAFE_KEY } from 'src/utils/constants'
 
-const buildOwnersFrom = (safeOwners: string[], localSafe?: SafeRecordProps): List<SafeOwner> => {
-  const ownersList = safeOwners.map((ownerAddress) => {
-    const convertedAdd = checksumAddress(ownerAddress)
+const extractSafeNativeBalance = (safeBalancesSettled: PromiseSettledResult<BalanceEndpoint>): string => {
+  let nativeCoinBalance = '0'
+  if (safeBalancesSettled.status === 'fulfilled') {
+    const safeBalances = safeBalancesSettled.value
+    const nativeCoin = safeBalances.items.find(({ tokenInfo }) => sameAddress(tokenInfo.address, ZERO_ADDRESS))
 
-    if (!localSafe) {
-      return makeOwner({ name: 'UNKNOWN', address: convertedAdd })
+    if (nativeCoin) {
+      nativeCoinBalance = fromTokenUnit(nativeCoin.balance, nativeCoin.tokenInfo.decimals)
     }
+  } else {
+    console.error('Failed to load Native coin balance information', safeBalancesSettled.reason)
+  }
+
+  return nativeCoinBalance
+}
 
-    const storedOwner = localSafe.owners.find(({ address }) => sameAddress(address, convertedAdd))
-    if (!storedOwner) {
-      return makeOwner({ name: 'UNKNOWN', address: convertedAdd })
+const extractRemoteSafeInfo = async (
+  remoteSafeInfoSettled: PromiseSettledResult<SafeInfo>,
+  checksummedSafeAddress: string,
+): Promise<Partial<SafeRecordProps>> => {
+  const safeInfo: Partial<SafeRecordProps> = {}
+
+  if (remoteSafeInfoSettled.status === 'fulfilled') {
+    const remoteSafeInfo = remoteSafeInfoSettled.value
+    const safeInfoModules = remoteSafeInfo.modules.map(({ value }) => value)
```

Comment:
Everything related to modules from here to line 61 could be skipped if the module list is empty.

Reply:
done! :male_detective: